src/core/WakaTimeCore.ts

```diff
@@ -135,14 +143,18 @@ class WakaTimeCore {
     ).heartbeat;
 
     const entity = settings.loggingType === 'domain' ? getDomainFromUrl(url) : url;
+
+    const projectNameFromList = this.getProjectNameFromList(url, settings);
+
     return {
       branch: heartbeat?.branch ?? '<<LAST_BRANCH>>',
       category: heartbeat?.category,
       entity: heartbeat?.entity ?? entity,
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-call
       id: uuid4(),
       language: heartbeat?.language,
       plugin: heartbeat?.plugin,
-      project: heartbeat?.project ?? '<<LAST_PROJECT>>',
+      project: projectNameFromList ?? heartbeat?.project ?? '<<LAST_PROJECT>>',
```

Comment:
We still need to use the project detected from DOM, which is in `heartbeat.project`

Reply:
So it should use defaults ?? in order project list, page, then last project.

Reply:
If the user has specified a custom project name for a given URL expression, shouldn't we ignore any name from the DOM or URL and use that instead?

Reply:
Sorry, somehow I didn't see `heartbeat?.project` still used there. All good!